### PR TITLE
BufReader: buffer size set to 1MB 

### PR DESCRIPTION
**Pull request by catevita**
BufReader wraps and impl Read struct in order to optimize the read() calls (calling less often), while caching the contents in an internal buffer. The default size of 8kb is quite tiny for point clouds, 1MB is a quite good limit for a buffer, especially in multi-threaded, multi-buffer implementation.

The benchmark test gives pretty much the same results on local disk, improvements should be noticed acquiring the data from remote.

### CODE REVIEW

**Comment by catevita** [comment]
PTAL @feuerste

**Comment by feuerste** [comment]
Can you please update the description of this PR and explain its reasoning a bit?

File: src/read_write/mod.rs [hunk 1]
```diff
@@ -38,3 +38,5 @@ pub use self::raw::{RawNodeReader, RawNodeWriter};
 
 mod s2;
 pub use self::s2::S2Splitter;
+
+pub const BUFFER_CAPACITY_1MB: usize = 1024 * 1024;
```
Comment (feuerste) [inline src/read_write/mod.rs]: How about just having a `pub const MB: u64 = 1024 * 1024;` in `src`, or, if it should be local to read/write, remove the `_1MB` suffix?

Comment (catevita) [reply src/read_write/mod.rs]: i changed it into pub const MB: usize = 1024 * 1024; in src

Comment (catevita) [reply src/read_write/mod.rs]: .. usize because capacity mostly requires usize type

File: src/read_write/ply.rs [hunk 2]
```diff
@@ -403,9 +403,11 @@ impl PlyIterator {
         }
 
         // We align the buffer of this 'BufReader' to points, so that we can index this buffer and know
-        // that it will always contain full points to parse.
+        // that it will always contain full points to parse. Capacity should be approximately 1MB
+        let num_points_in_mb: usize = BUFFER_CAPACITY / num_bytes_per_point;
```
Comment (feuerste) [inline src/read_write/ply.rs]: This should probably not use `BUFFER_CAPACITY`, as it may be different to one MB.

File: src/lib.rs [hunk 3]
```diff
@@ -88,5 +88,7 @@ pub struct PointsBatch {
 
 pub use point_viewer_proto_rust::proto;
 
-/// size for batch
-pub const NUM_POINTS_PER_BATCH: usize = 500_000;
+/// size for batch - with ca 24 bytes per point ~< 0.5 MB total
```
Comment (feuerste) [inline src/lib.rs]: I don't understand this comment, 24*128*1024 gives about 3.1 MB?

**Comment by catevita** [comment]
the modifications are not necessary anymore.

**Pull request closed without merging**